### ISSUE TITLE
Variable isn't visible inside include if it is not used in extended template

### ISSUE THREAD

**Issue by xen** [pallets/jinja#352]
Assume I have this structure:

``` jinja
<!-- nav.html -->
<div id="{{ nav_id|default('normal') }}">
...some html...
</div>
```

``` jinja
<!-- base.html -->
...skip...
{% include "nav.html" %}
```

and finally

``` jinja
{% extends "base.html" %}
{% set nav_id="index" %}
```

`nav_id` will not be visible in `nav.html` unless I access this variable in `base.html`. For example this way make `nav_id` work again in `nav.html`:

``` jinja
<!-- base.html -->
...skip...
<!-- {{ nav_id }} -->
{% include "nav.html" %}
```

As long as there is workaround it is not big problem, but still looks inconsistent.

**Comment by Ernest0x** [comment]
I can confirm this bug with jinja 2.8, but in my case the suggested workaround above does not work at all. It doesn't matter if the variable is used or not before the include statement. Evenmore, my case is even simpler as I do not use template inheritance. I have just a simple template with a for loop and an include statement inside the loop that includes another simple template that tries to access the variable set with each iteration in the loop. This seems to be a critical regression as things do not work as explained in the Note at http://jinja.pocoo.org/docs/dev/templates/#import-visibility

**Comment by Ernest0x** [comment]
> I can confirm this bug with jinja 2.8, but in my case the suggested workaround above does not work at all. It doesn't matter if the variable is used or not before the include statement. Evenmore, my case is even simpler as I do not use template inheritance. I have just a simple template with a for loop and an include statement inside the loop that includes another simple template that tries to access the variable set with each iteration in the loop. This seems to be a critical regression as things do not work as explained in the Note at http://jinja.pocoo.org/docs/dev/templates/#import-visibility

False alarm. I was using jinja through Ansible and it is Ansible's fault (it uses jinja in a way that included templates are not passed the context). It is fixed for the upcoming version (2.0) of Ansible.

**Comment by jackwilsdon** [comment]
+1 on this, I'm experiencing the same with a block and an extends:

#### layout.jinja2

```django
Hello, {% block world %}{% endblock %}
`````

#### my_page.jinja2

```django
{% set text = 'Pizza' %}
{% extends "layout.jinja2" %}
{% block world %}
  {% include "world_partial.jinja2" %}
{% endblock %}
```

#### world_partial.jinja2
```
{{ text }}!
```

Here is what I'd expect the output to be upon rendering `my_page.jinja2`:

```
Hello, Pizza!
```

But instead I get this:

```
Hello, {{ text }}!
```

---

You can "fix" it using @xen's method of using the variable inside the block:

#### my_page.jinja2

```django
{% set text = 'Pizza' %}
{% extends "layout.jinja2" %}
{% block world %}
  <!-- text is {{ text }} -->
  {% include "world_partial.jinja2" %}
{% endblock %}
```

But this just feels like a hack to me.

**Comment by jackwilsdon** [comment]
Oops, I guess that'll teach me to force push!

**Comment by ricky-undeadcoders** [comment]
This is not reproducible in version 2.10. It looks to have been solved with #619. 

Can this be closed?
